analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -187,14 +187,19 @@ abstract class PackageManager(
 
     /**
      * Return a tree of resolved dependencies (not necessarily declared dependencies, in case conflicts were resolved)
-     * for all [definitionFiles] which were found by searching the [analyzerRoot] directory.
+     * for all [definitionFiles] which were found by searching the [analyzerRoot] directory. By convention, the
+     * [definitionFiles] must be absolute.
      */
     open fun resolveDependencies(definitionFiles: List<File>): ResolutionResult {
         val result = mutableMapOf<File, ProjectAnalyzerResult>()
 
         prepareResolution(definitionFiles)
 
         definitionFiles.forEach { definitionFile ->
+            require(definitionFile.isAbsolute) {
```

Comment:
nit: Doing this in line 194, in particular before prepareResolution would be better in terms of 'fail early'.

Reply:
Can you please also 'require' the definitionFile to be located underneath the analyzerRoot ?

Reply:
> 
> 
> nit: Doing this in line 194, in particular before prepareResolution would be better in terms of 'fail early'.

I didn't want to add another `forEach` loop, but yes, maybe you're right that the check should be done before calling `prepareResolution()`.

Reply:
> 
> 
> Can you please also 'require' the definitionFile to be located underneath the analyzerRoot ?

I can probably combine these, let's see...